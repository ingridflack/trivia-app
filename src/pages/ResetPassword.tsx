import { Box, Card, Container, Image, Text } from "@chakra-ui/react";
import Header from "../components/Header";
import ResetPasswordForm from "../components/ResetPasswordForm";

export default function ResetPassword() {
  return (
    <>
      <Header />

      <Container
        display="flex"
        alignItems="center"
        justifyContent="center"
        minH="calc(var(--chakra-vh) - 50px)" // 100vh - 50px (height of the header)
      >
        <Card
          display="flex"
          flexDirection={["column", "column", "row"]}
          justifyContent="space-around"
          width="100%"
          padding="70px"
          shadow="xl"
        >
          <Box maxW="392px" width="100%">
            <Text fontSize="xx-large" marginBottom="32px" color="gray.700">
              Reset your password
            </Text>
            <ResetPasswordForm />
          </Box>

          <Image
            src="/src/assets/recovery.svg"
            alt="Login image"
            width="325px"
            height="324px"
            pointerEvents="none"
          />
        </Card>
      </Container>
    </>
  );
}
