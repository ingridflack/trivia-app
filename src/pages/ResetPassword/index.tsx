import { Box, Card, Container, Image, Text, useToast } from "@chakra-ui/react";
import Header from "../../components/Header";
import ResetPasswordForm from "../../components/ResetPasswordForm";
import { useNavigate, useParams } from "react-router-dom";
import { SubmitHandler, useForm } from "react-hook-form";
import * as AuthService from "../../services/authService";
import { Footer } from "../../components/Footer";

export interface ResetPasswordFormValues {
  password: string;
  passwordConfirmation: string;
}

export default function ResetPassword() {
  const {
    handleSubmit,
    register,
    formState: { errors, isSubmitting },
  } = useForm<ResetPasswordFormValues>();
  const toast = useToast();
  const navigate = useNavigate();
  const { token } = useParams();

  const onSubmit: SubmitHandler<ResetPasswordFormValues> = async (values) => {
    if (!token) return;

    try {
      await AuthService.resetPassword(token, values);

      toast({
        title: "Email sent.",
        description: "Check your inbox.",
        status: "success",
        duration: 2000,
        isClosable: true,
        position: "top-right",
      });

      navigate("/login");
    } catch (error) {
      console.log(error);
      toast({
        title: "An error occurred.",
        description: "Unable to login.",
        status: "error",
        duration: 2000,
        isClosable: true,
        position: "top-right",
      });
    }
  };
  return (
    <>
      <Header />

      <Container
        as="main"
        display="flex"
        alignItems="center"
        justifyContent="center"
        minH="calc(var(--chakra-vh) - 50px)" // 100vh - 50px (height of the header)
      >
        <Card
          display="flex"
          flexDirection={["column", "column", "row"]}
          justifyContent="space-around"
          width="100%"
          padding={{ base: "40px 20px", md: "80px 0" }}
          shadow="xl"
        >
          <Box maxW="392px" width="100%">
            <Text fontSize="xx-large" marginBottom="32px" color="gray.700">
              Reset your password
            </Text>
            <ResetPasswordForm
              onSubmit={handleSubmit(onSubmit)}
              errors={errors}
              register={register}
              isSubmitting={isSubmitting}
            />
          </Box>

          <Image
            src="/assets/recovery.svg"
            alt="Login image"
            width="325px"
            height="324px"
            pointerEvents="none"
            display={{ base: "none", md: "none", lg: "block" }}
          />
        </Card>
      </Container>
      <Footer />
    </>
  );
}
