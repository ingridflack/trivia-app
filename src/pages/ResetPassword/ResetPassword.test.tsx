import { describe, expect, it } from "vitest";
import ResetPassword from ".";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("ResetPassword", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <ResetPassword />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
