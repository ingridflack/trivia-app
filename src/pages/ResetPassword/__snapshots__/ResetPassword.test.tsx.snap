// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResetPassword > should match the snapshot 1`] = `
<div>
  <header
    class="css-cyjglw"
  >
    <div
      class="chakra-container css-1g360fb"
    >
      <div
        class="chakra-stack css-2jbh02"
      >
        <a
          href="/"
        >
          <p
            class="chakra-text css-kkaoep"
          >
            Trivia
          </p>
        </a>
        <div
          class="css-j0ozid"
        >
          <a
            href="/login"
          >
             Sign in 
          </a>
          <a
            href="/sign-up"
          >
             Sign Up 
          </a>
        </div>
      </div>
    </div>
  </header>
  <main
    class="chakra-container css-1vom740"
  >
    <div
      class="chakra-card css-61ehub"
    >
      <div
        class="css-1updlva"
      >
        <p
          class="chakra-text css-mvfx24"
        >
          Reset your password
        </p>
        <div>
          <p>
            Invalid token.
          </p>
        </div>
      </div>
      <img
        alt="Login image"
        class="chakra-image css-1b9ujuc"
        src="/assets/recovery.svg"
      />
    </div>
  </main>
  <footer
    class="css-1s2lyti"
  >
    <div
      class="chakra-container css-0"
    >
      <div
        class="chakra-stack css-11h4xmw"
      >
        <a
          href="/"
        >
          <p
            class="chakra-text css-rekzoj"
          >
            Trivia
          </p>
        </a>
        <div
          class="css-0"
        >
          <strong
            class="chakra-heading css-kf3i1p"
          >
            Site map
          </strong>
          <div
            class="chakra-stack css-1ko7mmx"
          >
            <a
              href="/"
            >
              Home
            </a>
            <a
              href="/login"
            >
              Sign in
            </a>
            <a
              href="/sign-up"
            >
              Sign up
            </a>
          </div>
        </div>
        <div
          class="css-0"
        >
          <strong
            class="chakra-heading css-kf3i1p"
          >
            Social
          </strong>
          <div
            class="chakra-stack css-1ko7mmx"
          >
            <a
              href="https://www.linkedin.com/in/ingrid-flack/"
            >
              LinkedIn
            </a>
            <a
              href="https://github.com/ingridflack"
            >
              GitHub
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1y3yl0q"
      >
        <p
          class="chakra-text css-0"
        >
          Â© 2024 Trivia App
        </p>
        <p
          class="chakra-text css-0"
        >
          Developed by Ingrid Flack
        </p>
      </div>
    </div>
  </footer>
</div>
`;
